x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -50,6 +49,20 @@ def config_conflict?
         false
       end
 
+      def pipeline_fetcher_factory
+        response = client.get("/")
+
+        if response["error"]
+          raise RemoteConfigError, "Cannot find elasticsearch version, server returned status: `#{response["status"]}`, message: `#{response["error"]}`"
+        end
+
+        logger.debug("Elasticsearch version ", response["version"]["number"])
+        version_number = response["version"]["number"].split(".")
+        first = version_number[0].to_i
+        second = version_number[1].to_i
+        (first >= 8 || (first == 7 && second >= 10))? SystemIndicesFetcher.new: LegacyHiddenIndicesFetcher.new
```

Comment:
Get the version number of elasticsearch and decide to go with system indices API or legacy API